### PR TITLE
Milestone 3: Set Up Associations

### PR DESCRIPTION
**Pull request by leopiresgoss**
## Milestone 3: Set Up Associations

In this milestone I:

- Created a class Classroom with the following: 
  - `@label` instance variable
  - Setter and getter for `@label`
- Created the `has-many`/`belongs-to` relationship between `Classroom` and `Student`
- Created a class `Book` with the following:
  - `@title` and `@author` instance variables
  - Setters and getters for instance variables
- Created a class `Rental` with the following:
  - `@date` instance variable  and its setter and getter
- Created the many-to-many (also has-many-through) relationship between `Person` and `Book` using the intermediate class `Rental`
- Added `app.rb` file where it was tested all relationships. You can test it by running: `ruby app.rb`

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Classroom
+  attr_accessor :label
+  attr_reader :students
+
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  def add_student(student)
```
Comment (FrederickMih) [inline classroom.rb:10]: [ ] When adding a student to a classroom, it should also set the classroom for the student: To achieve this, you need to add this code line.

```ruby
    def add_student(student)
         ....
         student.classroom = self
    end
```

- You may also link to [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships)

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (FrederickMih) [inline book.rb]: - [x] I think we can also modify this property. If that's the case please let's go like this: ` attr_accessor :title, :author, rentals`

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental)
```
Comment (FrederickMih) [inline book.rb]: - [x] In this method, I think we need to create a new Rental object, that's the person renting out the book and the book itself should be sent as a parameter achieving the `has-many` association. We may instead go like this: 

```ruby
  def add_rental(person, date)
   
      Rental.new(date, self, person)
  end
```

- [This article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) may be fruitful to you :+1:

File: person.rb [hunk 4]
```diff
@@ -25,6 +26,10 @@ def correct_name
     @name
   end
 
+  def add_rental(rental)
```
Comment (FrederickMih) [inline person.rb]: - [x] To create a new rental, achieving the `has-many` association, the person itself should be sent as a parameter. Let's check out this below:

```ruby
   def add_rental(book, date)
  
         Rental.new(date, book, self)
   end
```

- This [Link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) may be beneficial to you.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @leopiresgoss , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



# To Highlights

- GitHub workflow has been followed :heavy_check_mark: 
- Linters are set up correctly :heavy_check_mark: 
- Good commit messages :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

### Hi Leo, 
## Congratulations 🎉 Thank you for implementing all the requirements 🙏
### Please go ahead and merge the branch!

## Happy Coding 👏👏👏
#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**